api/v1alpha3/awsprincipal_types.go

```diff
@@ -0,0 +1,179 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type AWSClusterPrincipalSpec struct {
+	// AllowedNamespaces is used to identify which namespaces are allowed to use the principal from.
+	// Namespaces can be selected either using an array of namespaces or with label selector.
+	// An empty allowedNamespaces object indicates that AWSClusters can use this principal from any namespace.
+	// If this object is nil, no namespaces will be allowed (default behaviour, if this field is not provided)
+	// A namespace should be either in the NamespaceList or match with Selector to use the principal.
+	//
+	// +optional
+	AllowedNamespaces *AllowedNamespaces `json:"allowedNamespaces"`
+}
+
+type AllowedNamespaces struct {
+	// An nil or empty list indicates that AWSClusters cannot use the principal from any namespace.
+	//
+	// +optional
+	// +nullable
+	NamespaceList []string `json:"list"`
+
+	// AllowedNamespaces is a selector of namespaces that AWSClusters can
+	// use this ClusterPrincipal from. This is a standard Kubernetes LabelSelector,
+	// a label query over a set of resources. The result of matchLabels and
+	// matchExpressions are ANDed.
+	//
+	// An empty selector indicates that AWSClusters cannot use this
+	// AWSClusterPrincipal from any namespace.
+	// +optional
+	Selector metav1.LabelSelector `json:"selector"`
+}
+
+type AWSRoleSpec struct {
+	// The Amazon Resource Name (ARN) of the role to assume.
+	RoleArn string `json:"roleARN"`
+	// An identifier for the assumed role session
+	SessionName string `json:"sessionName,omitempty"`
+	// The duration, in seconds, of the role session before it is renewed.
+	// +kubebuilder:validation:Minimum:=900
+	// +kubebuilder:validation:Maximum:=43200
+	DurationSeconds int32 `json:"durationSeconds,omitempty"`
+	// An IAM policy in JSON format that you want to use as an inline session policy.
```

Comment:
```suggestion
	// An IAM policy as a JSON encoded string that you want to use as an inline session policy.
```

I know I wrote this in the proposal, but just realised,  we should clarify to users that it's still a string.